modules/loader.js

```diff
@@ -598,6 +560,92 @@ __private.loadBlockChain = function() {
 		});
 };
 
+/**
+ * Snapshot creation - performs rebuild of accounts states from blockchain data
+ *
+ * @private
+ * @emits snapshotFinished
+ * @throws {Error} When blockchain is shorter than one round of blocks
+ */
+__private.createSnapshot = height => {
+	library.logger.info('Snapshot mode enabled');
+
+	// Single round contains amount of blocks equal to number of active delegates
+	if (height < constants.activeDelegates) {
+		throw new Error(
+			'Unable to create snapshot, blockchain should contain at least one round of blocks'
+		);
+	}
+
+	const snapshotRound = library.config.loading.snapshotRound;
+	const totalRounds = Math.floor(height / constants.activeDelegates);
+	const targetRound = isNaN(snapshotRound)
+		? totalRounds
+		: Math.min(totalRounds, snapshotRound);
+	const targetHeight = targetRound * constants.activeDelegates;
+
+	library.logger.info(
+		`Snapshotting to end of round: ${targetRound}, height: ${targetHeight}`
+	);
+
+	let currentHeight = 1;
+	async.series(
+		{
+			resetMemTables(seriesCb) {
+				library.logic.account.resetMemTables(seriesCb);
+			},
+			loadBlocksOffset(seriesCb) {
+				async.until(
+					() => targetHeight < currentHeight,
+					untilCb => {
+						library.logger.info(
+							`Rebuilding accounts states, current round: ${slots.calcRound(
+								currentHeight
+							)}, height: ${currentHeight}`
+						);
+						modules.blocks.process.loadBlocksOffset(
+							constants.activeDelegates,
```

Comment:
@4miners We have a config `config.loading.loadPerIteration` which where used before as the `limit`. Is it a specific requirement to change it to `constants.activeDelegates`

Reply:
We need to always process only 1 round of blocks (101), so we can end snapshotting exactly after round ended.